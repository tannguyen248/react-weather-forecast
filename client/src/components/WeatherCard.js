import React from "react";
import Card from "react-bootstrap/Card";
import Temparature from "./Temparature";
import { URL_WEATHER_IMAGE } from "../constants/routes";
import DateOfMonth from "./DateOfMonth";

const WeatherCard = ({ stateAbbr, dayOfWeek, date, minTemp, maxTemp }) => {
  return (
    <Card className="text-center">
      <Card.Img variant="top" src={`${URL_WEATHER_IMAGE}${stateAbbr}.svg`} />
      <Card.Body>
        <Card.Title>
          {dayOfWeek} (
          <DateOfMonth date={date.getDate()} month={date.getMonth() + 1} />)
        </Card.Title>
        <Card.Text>
          <Temparature number={maxTemp} symbol="o" /> -{" "}
          <Temparature number={minTemp} symbol="o" />
        </Card.Text>
      </Card.Body>
    </Card>
  );
};

export default WeatherCard;
