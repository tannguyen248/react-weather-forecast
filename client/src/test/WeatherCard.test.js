import renderer from "react-test-renderer";
import WeatherCard from "../components/WeatherCard";

describe("WeatherCard", () => {
  it("should display weather card with data", () => {
    const stateAbbr = "c";
    const dayOfWeek = "1";
    const date = new Date("2020-01-01");
    const minTemp = 5;
    const maxTemp = 10;
    const tree = renderer
      .create(
        <WeatherCard {...{ stateAbbr, dayOfWeek, date, minTemp, maxTemp }} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
