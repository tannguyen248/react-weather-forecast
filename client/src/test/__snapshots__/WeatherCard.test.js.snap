// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should display weather card with data 1`] = `
<div
  className="text-center card"
>
  <img
    className="card-img-top"
    src="https://www.metaweather.com/static/img/weather/c.svg"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      1
       (
      <span>
        1
        -
        1
      </span>
      )
    </div>
    <p
      className="card-text"
    >
      <span>
        10
        <sup
          className="sc-bdfBwQ kqJIKH"
        >
          o
        </sup>
      </span>
       -
       
      <span>
        5
        <sup
          className="sc-bdfBwQ kqJIKH"
        >
          o
        </sup>
      </span>
    </p>
  </div>
</div>
`;
